CodeGenerator/CodeGenerator/FieldSerializer.cs

```diff
@@ -271,6 +271,23 @@ static void BytesWriter(Field f, string stream, CodeWriter cw)
             cw.WriteLine(stream + ".Write(msField.GetBuffer(), 0, (int)length" + f.ID + ");");
         }
 
+        private static void WriteOptionalFieldCheck( CodeWriter cw, Field f )
+        {
+            // Support fields with custom default values
+            string defaultString = f.OptionDefault ?? "default";
+
+            // For some reason Ray isn't comparible so just check it's fields 
+            if (f.ProtoTypeName == "Ray")
+            {
+                cw.IfBracket( $"instance.{f.CsName}.origin != default && instance.{f.CsName}.direction != default" );
+            }

```

Comment:
Were all the other Unity types in Common.proto tested? This could be generated rather than hard coded since all that info is there.

Reply:
Double check that it builds cause I got some random compile error trying to check if `jobHandle == default`. Only when building - not in the editor.